department-of-veterans-affairs/vets-website
# Import royalties and other properties chapter

## THIS FORM IS DISABLED IN PRODUCTION

# Income and Asset Statement Form (21P-0969)

## Description
Add a new section with the [Array Builder Pattern(Multiple responses list & loop)](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/patterns/array-builder/README.md)

## Issue
[Section VI: Income and net worth associated with royalties and other properties #86362](https://github.com/department-of-veterans-affairs/va.gov-team/issues/86362)

### Related Issues
n/a

### Epic
[Digitize Income and Asset Statement Form 21P-0969](https://app.zenhub.com/workspaces/benefits-pension-64e775aaa6b1ca1ed49b2ede/issues/gh/department-of-veterans-affairs/va.gov-team/80925)

### Screenshots
| Intro Page | Income Relationship Page | Generated Income Page | Summary Page |
| ------- |  ------- |  ------- |  ------- |
|![localhost_3001_income-and-asset-statement-form-21p-0969_ (14)](https://github.com/user-attachments/assets/9ee4949b-2ec2-4440-b0a9-64c3c8b54c87)|![localhost_3001_income-and-asset-statement-form-21p-0969_ (15)](https://github.com/user-attachments/assets/72de6cdd-d8da-4a94-9f81-ca4ee51be3f5)|![localhost_3001_income-and-asset-statement-form-21p-0969_ (16)](https://github.com/user-attachments/assets/4767db3a-0fa2-4666-ba34-e1300ca67aef)|![localhost_3001_income-and-asset-statement-form-21p-0969_ (17)](https://github.com/user-attachments/assets/256ab8bb-fa9f-4d00-9f0a-3cfc38f7715c)

## How to run in local environment
1. Check out this branch locally in 'vets-website'
2.  Checkout this [branch](https://github.com/department-of-veterans-affairs/content-build/pull/2075) locally in 'content-build'
3. Run 'vets-api' in your local environment
4. Run the 'vets-website' app with `yarn watch --env entry=static-pages,auth,login-page,profile,dashboard,income-and-asset-statement`

## How to toggle feature flag in local environment
1. Go to [income_and_assets_form_enabled](http://localhost:3000/flipper/features/income_and_assets_form_enabled)

## How to verify
1. Go to `http://localhost:3001/income-and-asset-statement-form-21p-0969` in your browser
2. Sign in with a mock user and 'Start application'
3. Go to the 'Royalties and other properties information' chapter
4. Verify that required fields are required
5. Verify that optional fields are optional
6. Verify that all fields follow validation rules
7. Verify that up to 5 royalties and other properties can be added, edited and deleted
8. Verify that the royalties and other properties are visible and editable on the 'Review and submit' page
9. Verify that the form can be saved via the api endpoint
10. Verify that the form can be submitted

## Testing

## What areas of the site does it impact?
Income and Asset Statement Application

### Quality Assurance & Testing

### Error Handling
- [ ] Events are being sent to the appropriate logging solution (if applicable)

### Authentication
